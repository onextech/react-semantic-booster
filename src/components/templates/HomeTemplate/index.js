import React from 'react';
import PropTypes from 'prop-types';
import Hero from '../../organisms/Hero';


const HomeTemplate = ({ children, heroProps }) => (
  <div>
    {heroProps && <Hero {...heroProps} />}
    {children}
  </div>
);

HomeTemplate.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.string,
    PropTypes.array,
  ]),
  heroProps: PropTypes.object,
};

HomeTemplate.defaultProps = {
  children: undefined,
  heroProps: undefined,
};

export default HomeTemplate;
