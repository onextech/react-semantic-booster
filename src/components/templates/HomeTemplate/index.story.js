import React from 'react';
import { storiesOf } from '@storybook/react';
import HomeTemplate from '.';
import { heroProps } from '../../organisms/Hero/index.story';
import Block from '../../atoms/Block';


storiesOf('HomeTemplate', module)
  .add('Default', () => (
    <HomeTemplate heroProps={heroProps}>
      <Block attached textAlign="center" inverted>
        Hello World
      </Block>
    </HomeTemplate>
  ));
