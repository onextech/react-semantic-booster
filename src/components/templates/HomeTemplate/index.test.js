import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import HomeTemplate from '.';
import { heroProps } from '../../organisms/Hero/index.story';


describe('<HomeTemplate>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Router><HomeTemplate heroProps={heroProps} /></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
