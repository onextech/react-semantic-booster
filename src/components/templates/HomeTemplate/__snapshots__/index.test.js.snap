// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeTemplate> renders correctly 1`] = `
<div>
  <div
    className="Hero__HeroFactory-hoaCgt hkHcUh"
    height={700}
    src="/"
  >
    <div
      className="ui text container"
    >
      <h1
        className="Hero__HeroHeader-dulUCc evQOct"
      >
        Header
      </h1>
      <p
        className="Hero__HeroSubheader-igSxRJ FLHjP"
      >
        Subheader
      </p>
      <a
        className="ui big primary button"
        href="#"
        onClick={[Function]}
        role="button"
        style={
          Object {
            "margin": "1em 0 0",
          }
        }
      >
        Btn.Content
          
        <i
          aria-hidden="true"
          className="arrow circle outline down icon"
        />
      </a>
    </div>
  </div>
</div>
`;
